File: src/main/resources/helloWorldClient.py
import grpc
import sys

import hello_pb2
import hello_pb2_grpc


def main():
    firstname = sys.argv[1] if len(sys.argv) > 1 else "Jonas.py"
    lastname = sys.argv[2] if len(sys.argv) > 2 else "Hösch"

    with grpc.insecure_channel("localhost:50051") as channel:
        stub = hello_pb2_grpc.HelloWorldServiceStub(channel)

        hello_request = hello_pb2.HelloRequest(firstname=firstname, lastname=lastname)
        hello_response = stub.hello(hello_request)
        print(hello_response.text)

        print("\nSending WarehouseRecord...\n")
        p1 = hello_pb2.ProductData(
            productId=101,
            productName="Laptop Pro 15",
            price=1499.99,
            quantity=3
        )

        p2 = hello_pb2.ProductData(
            productId=102,
            productName="Mouse Wireless",
            price=29.99,
            quantity=10
        )

        warehouse = hello_pb2.WarehouseRecord(
            warehouseID="W-001",
            warehouseName="Main Warehouse Europe",
            timestamp="2025-12-02T10:15:00Z",
            productDataList=[p1, p2]
        )

        warehouse_response = stub.sendWarehouse(warehouse)
        print("Server Response: " + warehouse_response.text)


if __name__ == "__main__":
    main()

File: src/main/java/HelloWorldClient.java
import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;

public class HelloWorldClient {

    public static void main(String[] args) {

        String firstname = args.length > 0 ? args[0] : "Jonas";
        String lastname = args.length > 1 ? args[1] : "Hösch";

        ManagedChannel channel = ManagedChannelBuilder.forAddress("localhost", 50051)
                .usePlaintext()
                .build();

        HelloWorldServiceGrpc.HelloWorldServiceBlockingStub stub = HelloWorldServiceGrpc.newBlockingStub(channel);

        Hello.HelloResponse helloResponse = stub.hello(Hello.HelloRequest.newBuilder()
                .setFirstname(firstname)
                .setLastname(lastname)
                .build());
        System.out.println(helloResponse.getText());



        System.out.println("\nSending WarehouseRecord...\n");

        // Build product list
        Hello.ProductData p1 = Hello.ProductData.newBuilder()
                .setProductId(101)
                .setProductName("Laptop Pro 15")
                .setPrice(1499.99)
                .setQuantity(3)
                .build();

        Hello.ProductData p2 = Hello.ProductData.newBuilder()
                .setProductId(102)
                .setProductName("Mouse Wireless")
                .setPrice(29.99)
                .setQuantity(10)
                .build();

        // Build warehouse record
        Hello.WarehouseRecord warehouse = Hello.WarehouseRecord.newBuilder()
                .setWarehouseID("W-001")
                .setWarehouseName("Main Warehouse Europe")
                .setTimestamp("2025-12-02T10:15:00Z")
                .addProductDataList(p1)
                .addProductDataList(p2)
                .build();

        // Call RPC
        Hello.HelloResponse warehouseResponse = stub.sendWarehouse(warehouse);
        System.out.println("Server Response: " + warehouseResponse.getText());

    }
}
